### PR TITLE
Use function annotation syntax for type hints.

### PR DESCRIPTION
**Pull request by bigbird555**
This PR is related to issue #711

### CODE REVIEW

**Comment by bigbird555** [comment]
If you do `$flake8 --ignore F401 logging.py`, there will be no errors on c25a2f8.

File: optuna/logging.py [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
 import logging
-from logging import CRITICAL  # NOQA
-from logging import DEBUG  # NOQA
-from logging import ERROR  # NOQA
-from logging import FATAL  # NOQA
-from logging import FATAL  # NOQA
-from logging import INFO  # NOQA
-from logging import WARN  # NOQA
-from logging import WARNING  # NOQA
+from logging import CRITICAL # NOQA
```
Comment (himkt) [inline optuna/logging.py]: I think you have to put two whitespaces behind the comments.

File: optuna/logging.py [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
-    from typing import Optional  # NOQA
+    from typing import Optional # NOQA
 
 _lock = threading.Lock()
 _default_handler = None  # type: Optional[logging.Handler]
 
 
-def create_default_formatter():
-    # type: () -> colorlog.ColoredFormatter
+def create_default_formatter() -> 'colorlog.ColoredFormatter':
```
Comment (himkt) [inline optuna/logging.py]:
```suggestion
def create_default_formatter() -> colorlog.ColoredFormatter:
```

We don't have to use forward reference since `colorlog` is imported in runtime.

File: optuna/logging.py [hunk 3]
```diff
@@ -30,20 +28,17 @@ def create_default_formatter():
...
-def _get_library_name():
-    # type: () -> str
+def _get_library_name() -> str:
 
     return __name__.split('.')[0]
 
 
-def _get_library_root_logger():
-    # type: () -> logging.Logger
+def _get_library_root_logger() -> 'logging.Logger':
```
Comment (himkt) [inline optuna/logging.py]: ditto

File: optuna/logging.py [hunk 4]
```diff
@@ -76,8 +70,7 @@ def _reset_library_root_logger():
         _default_handler = None
 
 
-def get_logger(name):
-    # type: (str) -> logging.Logger
+def get_logger(name: str) -> 'logging.Logger':
```
Comment (himkt) [inline optuna/logging.py]: ditto

**Comment by himkt** [comment]
Thank you for revising your PR, basically LGTM!
Same as #993, I sent you the PR for applying `black`, please check it.
https://github.com/bigbird555/optuna/pull/2

**Comment by bigbird555** [comment]
It seems CI still uses autopep8.....

**Comment by hvy** [comment]
Indeed it seems like your CirleCI configuration still invokes autopep8 (https://github.com/optuna/optuna/blob/6a0fc25c1a6a1f94594ab55a4e099e0661c01751/.circleci/config.yml). Could you update your CircleCI configuration to our latest master branch (https://github.com/optuna/optuna/blob/master/.circleci/config.yml)?

**Comment by bigbird555** [comment]
@hvy Okay. Thank you for your response.

**Comment by bigbird555** [comment]
@himkt In this PR, I have already incorporated CI change from autopep8 to black. Please check it out ! But, I still can not update CircleCI configuration in #993 and do not know why I can not....

File: optuna/logging.py [hunk 5]
```diff
@@ -1,12 +1,14 @@
 import colorlog
 import logging
+
```
Comment (hvy) [inline optuna/logging.py]: We usually cluster standard library imports in a single island. Could you undo this change?
```suggestion
```

File: optuna/logging.py [hunk 6]
```diff
@@ -1,12 +1,14 @@
...
 from logging import CRITICAL  # NOQA
 from logging import DEBUG  # NOQA
 from logging import ERROR  # NOQA
 from logging import FATAL  # NOQA
 from logging import INFO  # NOQA
 from logging import WARN  # NOQA
 from logging import WARNING  # NOQA
+
```
Comment (hvy) [inline optuna/logging.py]: Ditto.
```suggestion
```
